pandas/tools/merge.py

```diff
@@ -894,13 +896,21 @@ def get_result(self):
             if self.axis == 0:
                 new_data = com._concat_compat([x.values for x in self.objs])
                 name = com._consensus_name_attr(self.objs)
-                return Series(new_data, index=self.new_axes[0], name=name).__finalize__(self, method='concat')
+                if self._is_sp_series:
```

Comment:
Yes, you can do that if `self.axis==0`, but concatenation could happen along a different axis, so I thought the code would be easier to understand if the different cases were handled in roughly the same way.